NetCord/Gateway/GatewayClient.cs

```diff
@@ -17,71 +20,385 @@ public partial class GatewayClient : WebSocketClient, IEntity
+
+    /// <summary>
+    /// Sent when a guild scheduled event is deleted.
+    /// The inner payload is a <see cref="GuildScheduledEvent"/> object.
+    /// </summary>
     public event Func<GuildScheduledEvent, ValueTask>? GuildScheduledEventDelete;
+
+    /// <summary>
+    /// Sent when a user has subscribed to a guild scheduled event.
+    /// </summary>
     public event Func<GuildScheduledEventUserEventArgs, ValueTask>? GuildScheduledEventUserAdd;
+
+    /// <summary>
+    /// Sent when a user has unsubscribed from a guild scheduled event.
+    /// </summary>
     public event Func<GuildScheduledEventUserEventArgs, ValueTask>? GuildScheduledEventUserRemove;
+
+    /// <summary>
+    /// Sent when an integration is created.
+    /// The inner payload is an integration object with a set <see cref="GuildIntegrationEventArgs.GuildId"/>.
+    /// </summary>
     public event Func<GuildIntegrationEventArgs, ValueTask>? GuildIntegrationCreate;
+
+    /// <summary>
+    /// Sent when an integration is updated.
+    /// The inner payload is an integration object with a set <see cref="GuildIntegrationEventArgs.GuildId"/>.
+    /// </summary>
     public event Func<GuildIntegrationEventArgs, ValueTask>? GuildIntegrationUpdate;
+
+    /// <summary>
+    /// Sent when an integration is deleted.
+    /// </summary>
     public event Func<GuildIntegrationDeleteEventArgs, ValueTask>? GuildIntegrationDelete;
+
+    /// <summary>
+    /// Sent when a new invite to a channel is created. Only sent if the bot has the <see cref="Permissions.ManageChannels"/> permission for the relevant channel.
+    /// </summary>
     public event Func<GuildInvite, ValueTask>? GuildInviteCreate;
+
+    /// <summary>
+    /// Sent when an invite is deleted. Only sent if the bot has the <see cref="Permissions.ManageChannels"/> permission for the relevant channel.
+    /// </summary>
     public event Func<GuildInviteDeleteEventArgs, ValueTask>? GuildInviteDelete;
+
+    /// <summary>
+    /// Sent when a message is created. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// The inner payload is a message object with set <see cref="Message.GuildId"/>, <c>member</c> and <c>mentions</c> fields.
+    /// </summary>
     public event Func<Message, ValueTask>? MessageCreate;
+
+    /// <summary>
+    /// Sent when a message is updated. Ephemeral messages do not use the guild channel. Because of this, they are tied to the <see cref="GatewayIntents.DirectMessages"/> intent, and the message object won't include a <see cref="Message.GuildId"/> or <c>member</c>.
+    /// The inner payload is a message object with the same extra fields as <see cref="MessageCreate"/>. Updates may contain only a partial <see cref="Message"/> object (but will always contain an <c>ID</c> and <c>channel_id</c>).
```

Comment:
`Message` only exists with extra fields, so it should not be specified. Also, partial messages sent in the event are not handled by NetCord and the event is not invoked in this situation.